foxtrot-core/src/main/java/com/flipkart/foxtrot/core/reroute/ClusterRerouteManager.java

```diff
@@ -0,0 +1,172 @@
+package com.flipkart.foxtrot.core.reroute;
+
+import com.flipkart.foxtrot.common.Date;
+import com.flipkart.foxtrot.core.querystore.impl.ElasticsearchConnection;
+import com.flipkart.foxtrot.core.querystore.impl.ElasticsearchUtils;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.collect.Lists;
+import lombok.extern.slf4j.Slf4j;
+import org.elasticsearch.action.admin.cluster.node.info.NodesInfoRequest;
+import org.elasticsearch.action.admin.cluster.node.info.NodesInfoResponse;
+import org.elasticsearch.action.admin.cluster.reroute.ClusterRerouteRequest;
+import org.elasticsearch.action.admin.cluster.reroute.ClusterRerouteResponse;
+import org.elasticsearch.action.admin.indices.stats.IndicesStatsRequest;
+import org.elasticsearch.action.admin.indices.stats.IndicesStatsResponse;
+import org.elasticsearch.cluster.routing.allocation.command.MoveAllocationCommand;
+import org.elasticsearch.index.shard.ShardId;
+import org.joda.time.DateTime;
+import ru.vyarus.dropwizard.guice.module.installer.order.Order;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.*;
+
+/***
+ Created by mudit.g on Sep, 2019
+ ***/
+@Slf4j
+@Singleton
+@Order(20)
+public class ClusterRerouteManager {
+
+    private final ElasticsearchConnection connection;
+    private final ClusterRerouteConfig clusterRerouteConfig;
+
+    @Inject
```

Comment:
@santanusinha This is added to allocate hot shards from hot nodes equally to all nodes
This happens if a node gets restarted/added into the cluster few hours before midnight